orm/src/test/java/org/hibernate/search/test/query/SortTest.java

```diff
@@ -272,6 +278,67 @@ public void testResultOrderedByEmbeddedAuthorNameAscending() throws Exception {
 		tx.commit();
 	}
 
+	@Test
+	public void testCombinedQueryOnIndexWithSortFieldAndIndexToBeUninverted() throws Exception {
+		Transaction tx = fullTextSession.beginTransaction();
+
+		Query query = queryParser.parse( "name:Bill" );
+		FullTextQuery hibQuery = fullTextSession.createFullTextQuery( query, Plumber.class, BrickLayer.class );
+		Sort sort = new Sort( new SortField( "sortName", SortField.Type.STRING ) ); //ASC
+		hibQuery.setSort( sort );
+
+		@SuppressWarnings("unchecked")
+		List<Book> result = hibQuery.list();
+		assertNotNull( result );
+		assertThat( result ).onProperty( "name" )
+			.describedAs( "Expecting results from index with sort field and uninverted index in the correct sort order" )
+			.containsExactly( "Bill the brick layer", "Bill the plumber" );
+
+		tx.commit();
+	}
+
+	@Test
+	public void testQueryOnIndexSharedByEntityWithRequiredSortFieldAndEntityWithoutRaisesException() throws Exception {
+		Transaction tx = fullTextSession.beginTransaction();
+
+		Query query = queryParser.parse( "name:Bill" );
+		FullTextQuery hibQuery = fullTextSession.createFullTextQuery( query, Thatcher.class, BrickLayer.class );
```

Comment:
I hoped the method name would say it all :) The difference is that `BrickLayer` (not `Plumber`!) defines the sort field whereas `Thatcher` does not. Both entities are mapped to the same index ("index sharing") and this test aims at discovering this inconsistent configuration. I first thought that uninverting the index in that case still would work, but apparently it doesn't: sorting across the uninverted and the doc value fields doesn't give the right order.

I'll add a more descriptive comment.